### PR TITLE
Scala 2.13

### PR DESCRIPTION
**Pull request by martinpallmann**
tries to resolve softwaremill/tapir#130
- [x] drop support for scala 2.11
- [x] use milestone versions for 2.13 dependencies
- [x] find a sttp backend for 2.13 for the sttpClient tests
- [x] find a sttp backend for 2.13 for the serverTests
- [x] fix not found implicit for codec in tests
- [ ] maybe include examples in the build for 2.13
- [ ] maybe include playground in the build for 2.13

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -126,112 +123,119 @@ lazy val openapiCirceYaml: Project = (project in file("openapi/openapi-circe-yam
...
 lazy val sttpClient: Project = (project in file("client/sttp-client"))
-  .settings(commonSettings: _*)
+  .settings(commonSettings(): _*)
   .settings(
     name := "tapir-sttp-client",
     libraryDependencies ++= Seq(
-      "com.softwaremill.sttp" %% "core" % sttpVersion,
-      "com.softwaremill.sttp" %% "async-http-client-backend-fs2" % sttpVersion % "test"
+      "com.softwaremill.sttp" %% "core" % Versions.sttp
+//      "com.softwaremill.sttp" %% "async-http-client-backend-fs2" % Versions.sttp % "test"
```
Comment (martinpallmann) [inline build.sbt]: there is no async-http-client-backend-fs2 for 2.13 yet.

File: build.sbt [hunk 2]
```diff
@@ -126,112 +123,119 @@ lazy val openapiCirceYaml: Project = (project in file("openapi/openapi-circe-yam
...
 lazy val serverTests: Project = (project in file("server/tests"))
-  .settings(commonSettings: _*)
+  .settings(commonSettings(Seq(scala2_12)): _*)
   .settings(
     name := "tapir-server-tests",
     publishArtifact := false,
-    libraryDependencies ++= Seq("com.softwaremill.sttp" %% "async-http-client-backend-cats" % sttpVersion)
+    libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 12)) =>
+        Seq("com.softwaremill.sttp" %% "async-http-client-backend-cats" % Versions.sttp)
```
Comment (martinpallmann) [inline build.sbt]: there is no async-http-client-backend-cats for 2.13 yet. What should I do here?

File: tests/src/main/scala/tapir/tests/package.scala [hunk 3]
```diff
@@ -12,6 +12,8 @@ import scala.io.Source
 
 package object tests {
 
+  implicit val fruitAmountCodec: Codec[FruitAmount, MediaType.XWwwFormUrlencoded, _] = ???
```
Comment (martinpallmann) [inline tests/src/main/scala/tapir/tests/package.scala]: For some reason the implicit is not found on scala 2.13. Maybe someone who knows more about implicit and magnolia and stuff could help here.

File: build.sbt [hunk 4]
```diff
@@ -1,39 +1,36 @@
...
-        )
-      case _ =>
-        Nil
-    })
-)
+val scala2_13 = "2.13.0"
+def commonSettings(scalaVersions: Seq[String] = Seq(scala2_12, scala2_13)) =
+  commonSmlBuildSettings ++ ossPublishSettings ++ Seq(
+    organization := "com.softwaremill.tapir",
+    scalaVersion := scala2_13,
```
Comment (adamw) [inline build.sbt]: shouldn't this use the first (?) scala version from `scalaVersions`?

File: tests/src/main/scala/tapir/tests/package.scala (hunk 3 above)
Comment (adamw) [reply tests/src/main/scala/tapir/tests/package.scala]: In `core`s tests, I'm getting a compile error for the tests:

```
[error] /Users/adamw/projects/tapir/core/src/test/scala/tapir/generic/FormCodecDerivationTest.scala:143:27: type mismatch;
[error]  found   : scala.collection.MapView[String,Seq[String]]
[error]  required: Map[String,Seq[String]]
[error]     val codec = implicitly[Codec[Test1, MediaType.XWwwFormUrlencoded, String]]
```

probably sth collection-related in the macro ...

Comment (adamw) [reply tests/src/main/scala/tapir/tests/package.scala]: Yeah, mapValues returns a MapView, an explicit conversion to Map is needed

In `FormCodecMacros:47`, you'll need to add a `.toMap` at the end of the line:

```
          val paramsMap: Map[String, Seq[String]] = params.groupBy(_._1).mapValues(_.map(_._2)).toMap
```

File: build.sbt (hunk 4 above)
Comment (martinpallmann) [reply build.sbt]: IDK, can do that.

File: build.sbt (hunk 2 above)
Comment (adamw) [reply build.sbt]: I copied your approach to sttp and released 1.6.3 with almost all modules for 2.13 (except monix & circe-js): https://github.com/softwaremill/sttp/commit/8457960c837aed8a73154e5845bd4ebf7d2aa633

So I think this should help? :)

Comment (martinpallmann) [reply build.sbt]: great.

**Comment by martinpallmann** [comment]
So, one problem which I don't know how to tackle. I didn't find a way to exclude the examples and playground from the 2.13 build. Since they depend on zoo-interop-cats, which is only released for 2.12, I'm stuck here.

**Comment by adamw** [comment]
@martinpallmann ok, thanks, I'll take it from here - I'll use the same hack as in sttp :)

**Comment by adamw** [comment]
See: https://github.com/softwaremill/tapir/commit/257f5db46e7872507cbfd53620c0836806c57759

**Comment by adamw** [comment]
Released in 0.9.0, thanks!

**Pull request merged**